hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -375,56 +385,117 @@ public Vertex updateProcessor(
+        RexNode rexPredicate = predicate == null ? null : predicate.unwrap(RexNode.class);
 
-        DeleteQueryBuilder builder = new DeleteQueryBuilder(table, pkFields, resolveDialect(table, context));
-        return context.getDag().newUniqueVertex(
-                "Delete(" + table.getExternalNameList() + ")",
-                new DeleteProcessorSupplier(
-                        table.getDataConnectionName(),
-                        builder.query(),
-                        table.getBatchLimit()
+        DeleteQueryBuilder builder = new DeleteQueryBuilder(table,
+                resolveDialect(table, context),
+                rexPredicate,
+                hasInput
+        );
+
+        DMLProcessorSupplier deletePS = new DMLProcessorSupplier(
+                table.getDataConnectionName(),
+                builder.query(),
+                builder.dynamicParams(),
+                builder.inputRefs(),
+                table.getBatchLimit()
+        );
+
+        return dmlVertex(context, hasInput, table, deletePS, "Delete");
+    }
+
+    private static Vertex dmlVertex(
+            DagBuildContext context,
+            boolean hasInput,
+            JdbcTable table,
+            DMLProcessorSupplier processorSupplier,
+            String statement
+    ) {
+
+        if (!hasInput) {
+            // There is no input, we push the whole update query to the database, but we need single fake item
+            // to execute the update in WriteJdbcP
+            // We can consider refactoring the WriteJdbcP, so it doesn't need the fake item in the future.
+
+            // Create random partition key so the fake source and processor run on the same member
+            String partitionKey = UuidUtil.newUnsecureUuidString();
+
+            Vertex v = fakeSourceVertex(context, "FakeSourceFor" + statement, partitionKey);
+            Vertex updateVertex = context.getDag().newUniqueVertex(
+                    statement + "(" + table.getExternalNameList() + ")",
+                    forceTotalParallelismOne(processorSupplier, partitionKey)
+            );
+
+            context.getDag().edge(Edge.between(v, updateVertex));
+            return updateVertex;
+        } else {
+            Vertex updateVertex = context.getDag().newUniqueVertex(
+                    statement + "(" + table.getExternalNameList() + ")",
+                    processorSupplier
+            ).localParallelism(1);
+            return updateVertex;
+        }
+    }
+
+    private static Vertex fakeSourceVertex(DagBuildContext context, String name, String partitionKey) {
+        Vertex v = context.getDag().newUniqueVertex(name,
+                forceTotalParallelismOne(
+                        of(() -> new ConvenientSourceP<>(
+                                ExpressionEvalContext::from,
+                                (evalContext, buf) -> {
+                                    SourceBuffer<Object> buffer = (SourceBuffer<Object>) buf;
```

Comment:
to check: how will it behave with processing guarantees? For batch jobs they are not working fully anyway, but would be nice know what will happen in such case